### PR TITLE
Feat/admin domains

### PR DESCRIPTION
**Pull request by shadowgate15**

### CODE REVIEW

File: app/controllers/web/admin/domains.js [hunk 1]
```diff
@@ -44,6 +45,27 @@ async function list(ctx) {
   ctx.body = { table };
 }
 
+async function remove(ctx) {
+  const domain = await Domains.findById(ctx.params.id);
+
+  if (!domain) throw Boom.notFound(ctx.translateError('INVALID_USER'));
```
Comment (niftylettuce) [inline app/controllers/web/admin/domains.js]: You're using the wrong phrase here, it should be `'INVALID_DOMAIN'`

File: app/views/admin/domains/_table.pug [hunk 2]
```diff
@@ -26,7 +26,14 @@ include ../../_pagination
...
               = t(titleize(humanize(domain.plan)))
             td.align-middle.text-center
-              = domain.max_recipients_per_alias
+              form.ajax-form.confirm-prompt.d-inline-block(action=l(`/admin/domains/${domain.id}`), method="POST", autocomplete="off")
+                input(type="hidden", name="_csrf", value=ctx.csrf)
+                input(type="hidden", name="_method", value="PUT")
+                .input-group
+                  select#input-max-recipients-per-alias(name='max_recipients_per_alias').form-select.form-select-sm
+                    each option in OPTIONS_MAX_RECIPIENTS
+                      option(value=option selected=option === domain.max_recipients_per_alias)= option
```
Comment (niftylettuce) [inline app/views/admin/domains/_table.pug:35]: You're missing a comma in between `value=option selected=option` (our code style).  Not sure if pug linter would have caught this, but I'd fix this and possibly skim through anywhere else you might forget it.

Comment (niftylettuce) [reply app/views/admin/domains/_table.pug:35]: Also there could be an arbitrary value here in the database so it should be an `input(type="number", inputmode="numeric", step=10, ..., value=domain.max_recipients_per_alias)`

File: app/controllers/web/admin/domains.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  if (ctx.accepts('html'))
+    return ctx.render('admin/domains', {
+      domains,
+      pageCount,
+      itemCount,
+      pages: paginate.getArrayPages(ctx)(3, pageCount, ctx.query.page)
+    });
+
+  // this will assign rendered html to ctx.body
+  await ctx.render('admin/domains/_table', {
```
Comment (niftylettuce) [inline app/controllers/web/admin/domains.js:38]: This is an anti-pattern and bad practice for code readability to assume users know that calling `ctx.render` assigns to `ctx.body`.

Instead you should write it out here as `ctx.body = await ctx.render(...)`.

I think all we'd have to do is add a `return html` here to `koa-views` to support this @ https://github.com/queckezz/koa-views/blob/master/src/index.js#L72

I'm an author on `koa-views` on npm I think so once you submit a PR I can merge and release with semver major bump.

Comment (niftylettuce) [reply app/controllers/web/admin/domains.js:38]: We will also need to change this everywhere else we do this practice, as it's an anti-pattern to write in a way that's not readable to newbies.

**Review by niftylettuce — CHANGES_REQUESTED** [review]
See comments, thank you!

File: app/controllers/web/admin/domains.js (hunk 3 above)
Comment (niftylettuce) [reply app/controllers/web/admin/domains.js:38]: See more context here <https://titanism.slack.com/archives/C02FU8YN0L8/p1637614211010500>

File: app/controllers/web/admin/domains.js (hunk 1 above)
Comment (shadowgate15) [reply app/controllers/web/admin/domains.js]: oh i thought i caught these. also it isn't [`INVALID_DOMAIN`](https://github.com/forwardemail/forwardemail.net/blob/feat/admin-domains/config/phrases.js#L55) it is  [`DOMAIN_DOES_NOT_EXIST`](https://github.com/forwardemail/forwardemail.net/blob/05b44b2ddb3a4c2e74ff2479f1b5d487a9c42118/config/phrases.js#L62)

Comment (shadowgate15) [reply app/controllers/web/admin/domains.js]: this is resolved in another commit.

File: app/views/admin/domains/_table.pug (hunk 2 above)
Comment (shadowgate15) [reply app/views/admin/domains/_table.pug:35]: oh i forgot about `step=`.... that is much better.

File: app/controllers/web/admin/domains.js (hunk 3 above)
Comment (shadowgate15) [reply app/controllers/web/admin/domains.js:38]: https://github.com/queckezz/koa-views/pull/161

waiting on this to be accepted.

**Pull request merged**